src/openloco/windows/promptbrowsewnd.cpp

```diff
@@ -300,11 +312,59 @@ namespace openloco::ui::prompt_browse
     }
 
     // 0x00445C8F
-    static void prepare_draw(ui::window* window)
+    static void prepare_draw(ui::window* self)
     {
+        // TODO: replace with a fixed length!
+        char* buffer = (char*)stringmgr::get_string(string_ids::buffer_2039);
+        strcpy(buffer, _title);
+
+        self->widgets[widx::frame].right = self->width - 1;
+        self->widgets[widx::frame].bottom = self->height - 1;
+
+        self->widgets[widx::panel].right = self->width - 1;
+        self->widgets[widx::panel].bottom = self->height - 1;
+
+        self->widgets[widx::caption].right = self->width - 2;
+
+        self->widgets[widx::close_button].left = self->width - 15;
+        self->widgets[widx::close_button].right = self->width - 3;
+
+        if (*_type == (uint8_t)browse_file_type::unk_2)
+        {
+            self->widgets[widx::ok_button].top = self->height - 16;
+            self->widgets[widx::ok_button].bottom = self->height - 5;
+
+            self->widgets[widx::text_filename].right = self->width - 6;
+            self->widgets[widx::text_filename].top = self->height - 32;
+            self->widgets[widx::text_filename].bottom = self->height - 19;
+
+            self->widgets[widx::text_filename].bottom = self->height - 35;
+
+            self->widgets[widx::ok_button].type = widget_type::wt_11;
+            self->widgets[widx::text_filename].type = widget_type::wt_17;
+        }
+        else
+        {
+            self->widgets[widx::text_filename].bottom = self->height - 4;
+
+            self->widgets[widx::ok_button].type = widget_type::none;
+            self->widgets[widx::text_filename].type = widget_type::none;
+        }
+
+        self->widgets[widx::scrollview].right = self->width - 261;
+        if (*_fileType != (uint8_t)browse_file_type::saved_game)
+            self->widgets[widx::scrollview].right += 122;
+
+        self->widgets[widx::text_filename].left = self->width - 86;
+        self->widgets[widx::text_filename].right = self->width - 16;
+
+        self->widgets[widx::parent_button].left = self->width - 26;
+        self->widgets[widx::parent_button].right = self->width - 3;
+
         registers regs;
-        regs.esi = (int32_t)window;
-        call(0x00445C8F, regs);
+        regs.edi = (int32_t)buffer;
+        regs.esi = (int32_t)self;
+        call(0x00445D91, regs);
```

Comment:
Can you add a comment to explain what this does?

Reply:
Will do. It just resumes the original `prepare_draw` routine, past the point of widget repositioning.